src/app-layer-parser.c

```diff
@@ -130,9 +130,15 @@ struct AppLayerParserState_ {
 
     /* State version, incremented for each update.  Can wrap around. */
     uint8_t version;
+
+    /* Indicates which loggers that have already logged the transaction,
+       using the logger id (power of two) as a bitwise flag. */
+    uint32_t logged;
```

Comment:
this doesn't track things per tx, which I think it should. The AppLayerParserState is per flow, not per tx.